### PR TITLE
Fix: [CMake] Check nforenum and grfcodec return value

### PR DESCRIPTION
**Pull request by glx22**
## Motivation / Problem
When `GRFCODEC_EXECUTABLE` and `NFORENUM_EXECUTABLE` are manually specified, they are considered valid and `GRFCODEC_FOUND` is set, but the variables could contain errors and execution silently fail later.

## Description
~~Always try to run `GRFCODEC_EXECUTABLE` and `NFORENUM_EXECUTABLE` and discard invalid values.~~
Check return value when running nforenum and grfcodec, and force the build to fail if needed.
I also fixed a dependency check error (modifying a .nfo would not trigger a grf rebuild)

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

**Comment by TrueBrain** [comment]
Honestly, I am not sure this is the correct solution. If you manually supply stuff, it should just try to execute that, and not evaluate if it is correct. For me it is such a slippery slope .. should we also check if it is really grfcodec, and not some other binary? :D Where does this end ;)
Also, I believe if you manually supply library locations, CMake also takes you on your word.

But aren't you running into a completely different problem, one I am having currently: if grfcodec runtime fails, CMake continues to build OpenTTD. Which is very .. euh .. annoying.

Example: change the sprite-count in `openttdgui.nfo` to one lower than you define. Now compile OpenTTD. It will succeed. But if you look closer, you see grfcodec failed, and the `openttdgui.grf` is an invalid file now.

Maybe if we solve that issue, the reason you created this issue is also resolved?

tldr, for me the keyword in your PR is `and execution silently fail later`, which to me is the real bug. This solution only fixes some parts of that real bug, still leaving a lot of room for things to go wrong another way :)

File: cmake/scripts/CreateGRF.cmake [hunk 1]
```diff
@@ -39,6 +39,20 @@ foreach(NFO_LINE IN LISTS NFO_LINES)
     endif()
 endforeach()
 
-execute_process(COMMAND ${NFORENUM_EXECUTABLE} -s sprites/${GRF_SOURCE_FOLDER_NAME}.nfo)
-execute_process(COMMAND ${GRFCODEC_EXECUTABLE} -n -s -e -p1 ${GRF_SOURCE_FOLDER_NAME}.grf)
+execute_process(COMMAND ${NFORENUM_EXECUTABLE} -s sprites/${GRF_SOURCE_FOLDER_NAME}.nfo RESULT_VARIABLE RESULT)
+if(RESULT)
+    if(NOT RESULT MATCHES "^[0-9]*$")
```
Comment (TrueBrain) [inline cmake/scripts/CreateGRF.cmake:44]: what does this regex capture? Maybe a comment of some kind might help future-us, but mainly I am just curious :D

Comment (glx22) [reply cmake/scripts/CreateGRF.cmake:44]: `RESULT` can be "an integer return code from the last child or a string describing an error condition" as stated in CMake documentation. So if it's not an integer it didn't run at all and there's probably an issue with the executable itself.

Comment (TrueBrain) [reply cmake/scripts/CreateGRF.cmake:44]: Smart :D

**Review by LordAro — APPROVED** [review]
